File: webcomponents/signal/signal.js
export function signal(initialValue){
    
    let value = initialValue
    let suscriptors = new Set();
    
    const fn = ()=>value   //gettter

    fn.set = (v)=>{  //setter
        if(!Object.is(value,v)){
            value=v
            for(const fn of suscriptors){
                fn();
            }
        }        
    }
    fn.subscribe = (fn)=>{ //suscription
        fn();
        suscriptors.add(fn)
    }
    return fn
}

File: python/signal.py
def signal(initialValue):
    state=initialValue
    suscriptors = []
    fn=lambda:state

    def set(v):        
        nonlocal state       
        if(v!=state):
            state=v       
            for cb in suscriptors:
                cb()

    fn.set = set
    
    def suscribe(cb):
        cb()
        suscriptors.append(cb)
    fn.suscribe = suscribe
    return fn

result = signal(10)
result.suscribe(lambda:print(result()))
result.set(100)
